api/app_analytics/serializers.py

```diff
@@ -12,6 +12,7 @@ class UsageDataSerializer(serializers.Serializer):
 class UsageDataQuerySerializer(serializers.Serializer):
     project_id = serializers.IntegerField(required=False)
     environment_id = serializers.IntegerField(required=False)
+    period = serializers.CharField(required=False)
```

Comment:
Do you think it's a good idea to use [RegexField](https://www.django-rest-framework.org/api-guide/fields/#regexfield) here?

Reply:
Done. Used the constants to be extra 👌🏼

Reply:
> Used the constants

So the period is not meant to be arbitrary? If we're using a fixed set of pre-defined periods, I'd suggest using ChoiceField instead.